lib/segment/src/index/field_index/index_selector.rs

```diff
@@ -81,7 +81,14 @@ impl<'a> IndexSelector<'a> {
             PayloadSchemaParams::Datetime(_) => {
                 vec![FieldIndex::DatetimeIndex(self.numeric_new(field)?)]
             }
-            PayloadSchemaParams::Uuid(_) => vec![FieldIndex::UuidIndex(self.numeric_new(field)?)],
+            PayloadSchemaParams::Uuid(_) => match self {
+                IndexSelector::RocksDb(_) => {
+                    vec![FieldIndex::UuidIndex(self.numeric_new(field)?)]
+                }
+                IndexSelector::OnDisk(_) => {
+                    vec![FieldIndex::UuidMapIndex(self.map_new(field)?)]
+                }
```

Comment:
I think there was a misunderstanding. I thought that by hybrid index we mean the same approach we use in `int` index, where we use numeric for ranges and map for match.

This approach creates numeric index, this will work for ranges _partially_.

Some segments might be on disk, others are not, so the range request will be executed only on some segments. I don't think user will expect that.

I will submit a review commit, which will only allow `UuidMapIndex` in all cases.
Initially I though to include numeric option as well, but it can be done later and won't give us much benefits, cause we don't have UUID support in range request yet anyway.